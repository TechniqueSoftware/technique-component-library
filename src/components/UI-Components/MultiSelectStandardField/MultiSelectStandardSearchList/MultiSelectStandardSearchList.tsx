import * as React from 'react';
import { Theme } from '@material-ui/core';
import List from '@material-ui/core/List';
import withStyles from '@material-ui/styles/withStyles';

import MultiSelectStandardCheckboxListItem, { MultiSelectCheckboxListItemProps } from '../MultiSelectStandardCheckboxListItem/MultiSelectCheckboxListItem';

const styles = (theme: Theme) => ({
  container: {
    border: `1px solid ${theme.palette.grey['300']}`,
    background: theme.palette.common.white
  },
  whiteBackground: {
    background: theme.palette.common.white
  }
});

export interface MultiSelectStandardOptions {
  name: string;
  labelString: string;
  value: string;
  checked: boolean;
}
export interface MultiSelectSearchListProps {
  classes?: any;
  listContainerHeight?: number;
  selectMaxWidth?: number;
  multiSelectStandardFieldProps?: MultiSelectStandardOptions[];
  onCheckboxGroupChange: (groupValues: MultiSelectStandardOptions[]) => void;
  disabled?: boolean;
}
export class MultiSelectStandardSearchList extends React.Component<MultiSelectSearchListProps> {

  static defaultProps: Partial<MultiSelectSearchListProps> = {
    listContainerHeight: 450,
    disabled: false
  };

  constructor(props: any) {
    super(props);
    this.handleCheckboxGroupChange = this.handleCheckboxGroupChange.bind(this);
  }

  handleCheckboxGroupChange(eventData: MultiSelectStandardOptions[]) {
    this.props.onCheckboxGroupChange(eventData);
  }

  renderCheckboxListItemWithHeader(item : any) {
    const { labelString, name, value, checked } = item;
    const { disabled, multiSelectStandardFieldProps } = this.props;

    const headerProps: MultiSelectCheckboxListItemProps = {
      labelString,
      value,
      checked,
      disabled,
      multiSelectStandardFieldProps,
      name: `${name}-allSelected`,
      onClick: this.handleCheckboxGroupChange,
    };

    return (
      <React.Fragment key={`checkboxList-${labelString}`}>
        <MultiSelectStandardCheckboxListItem {...headerProps} />
      </React.Fragment>
    );
  }

  render() {
    const { classes, listContainerHeight, selectMaxWidth, multiSelectStandardFieldProps } = this.props;
    const listStyles = { overflow: 'auto', maxHeight: listContainerHeight, width: selectMaxWidth };
    return (
      <div className={classes.container}>
        <List subheader={<li />} style={listStyles} dense>
          {multiSelectStandardFieldProps
            && multiSelectStandardFieldProps.map((item: any) => {
              return this.renderCheckboxListItemWithHeader(item);
            })}
        </List>
      </div>
    );
  }
}

export default withStyles(styles)(MultiSelectStandardSearchList);
