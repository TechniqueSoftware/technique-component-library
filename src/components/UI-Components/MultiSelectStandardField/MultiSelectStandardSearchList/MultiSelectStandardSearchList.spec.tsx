import { shallow } from 'enzyme';
import * as React from 'react';
import MultiSelectStandardSearchList, { MultiSelectSearchListProps } from './MultiSelectStandardSearchList';
import { MuiThemeProvider } from '@material-ui/core/styles';
import clubOsTheme from '../../storyThemes';

describe('MultiSelectSearchList', () => {
  const baseProps: MultiSelectSearchListProps = {
    classes: {
      searchContainer: 'mockSearchContainerClass'
    },
    listContainerHeight: 400,
    selectMaxWidth: 300,
    multiSelectStandardFieldProps: [
      {
        labelString: 'First-Time Leads',
        value: 'FirstTimeLeads',
        name: 'FirstTimeLeads',
        checked: true
      },
      {
        labelString: 'Re-Interested Leads',
        value: 'ReInterestedLeads',
        name: 'ReInterestedLeads',
        checked: true
      },
      {
        labelString: 'Re-Interested Ex-Members',
        value: 'ReInterestedExMembers',
        name: 'ReInterestedExMembers',
        checked: false
      }
    ],
    onCheckboxGroupChange: jest.fn(),
    disabled: false,
  };

  test('should render with minimal props', () => {
    const props = { ...baseProps };
    const shallowWrapper = shallow(
      <MuiThemeProvider theme={clubOsTheme} >
        <MultiSelectStandardSearchList {...props} />
      </MuiThemeProvider>
    );
    const searchInputWrapper = shallowWrapper.find('WithStyles(TextInput)');
    expect(searchInputWrapper.debug()).toMatchSnapshot();
  });
});
