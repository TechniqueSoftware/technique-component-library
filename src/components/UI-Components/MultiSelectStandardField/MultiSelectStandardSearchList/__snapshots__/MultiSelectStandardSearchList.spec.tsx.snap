// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiSelectSearchList should render with minimal props 1`] = `""`;
